### PR TITLE
[INLONG-11013][Sort] A demo of reporting sort logs through OpenTelemetry

### LINKED ISSUE #11013: [Feature][Sort] A demo of reporting sort logs through OpenTelemetry
### Description

- parent issue #10798 

Make a simple demo of eporting sort logs through OpenTelemetry
Implement the function of uploading connector logs to opentelemetry-collector

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qy-liuhuo**
Fixes #11013 

### Motivation

- Make a simple demo of eporting sort logs through OpenTelemetry for next integration work

### Modifications

Implement the demo of uploading mysql-connector logs to opentelemetry-collector

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

### CODE REVIEW

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml [hunk 1]
```diff
@@ -20,22 +20,73 @@
...
+            <version>${log4j2.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry.instrumentation</groupId>
+            <artifactId>opentelemetry-instrumentation-api</artifactId>
+            <version>1.28.0</version>
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml:52]: add a new property

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml [hunk 2]
```diff
@@ -20,22 +20,73 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry.instrumentation</groupId>
+            <artifactId>opentelemetry-log4j-appender-2.17</artifactId>
+            <version>1.28.0-alpha</version>
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml:57]: ditto

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml [hunk 3]
```diff
@@ -20,22 +20,73 @@
...
+            <artifactId>opentelemetry-exporter-otlp</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-sdk</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-semconv</artifactId>
+            <version>1.28.0-alpha</version>
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml:74]: ditto

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml [hunk 4]
```diff
@@ -20,22 +20,73 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-sdk-logs</artifactId>
+            <version>1.28.0</version>
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml:79]: ditto

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml [hunk 5]
```diff
@@ -20,22 +20,73 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.squareup.okio</groupId>
+            <artifactId>okio</artifactId>
+            <version>1.17.2</version>
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml:88]: ditto

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/mysql/source/reader/MySqlSourceReader.java [hunk 6]
```diff
@@ -88,6 +95,7 @@ public class MySqlSourceReader<T>
     private final MySqlSourceReaderContext mySqlSourceReaderContext;
     private MySqlBinlogSplit suspendedBinlogSplit;
     private final DebeziumDeserializationSchema<T> metricSchema;
+    private OpenTelemetrySdk SDK;
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/mysql/source/reader/MySqlSourceReader.java:98]: Better to add a configuration to enable or disable opentelemetry

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml (hunk 1 above)
Comment (qy-liuhuo) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/pom.xml:52]: I found a better way to do this #11065 .

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/mysql/source/reader/MySqlSourceReader.java (hunk 6 above)
Comment (qy-liuhuo) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/mysql/source/reader/MySqlSourceReader.java:98]: I found a better way to do this #11065 .

**Comment by aloyszhang** [comment]
@qy-liuhuo do we still need to keep this pull request open?

**Comment by qy-liuhuo** [comment]
> @qy-liuhuo do we still need to keep this pull request open?

As there is a new solution, this PR should be closed. Thanks

**Pull request closed without merging**